osu.Game/Rulesets/UI/ReplayRecorder.cs

```diff
@@ -39,6 +49,14 @@ protected override void LoadComplete()
             base.LoadComplete();
 
             inputManager = GetContainingInputManager();
+
+            spectatorStreaming?.BeginPlaying();
```

Comment:
Just going to point out that attaching the spectator client like this means all people playing lazer will be sending frames to the spectator server. Which I guess isn't much of an issue even if the server gets accidentally ddos'd due to that, as the test scene seems to be unaffected by the spec server dying under it, but maybe something to keep in mind...

I spotted this as visual tests opened on another test (autoplay, to be exact) and I started seeing incoming frames on the server component.

Reply:
this is intentional. this will also likely be how replays are stored server-side.

it shouldn't be firing on autoplay though, worth investigation.